src/main/java/org/mule/extensions/java/internal/parameters/ConstructorIdentifier.java

```diff
@@ -70,7 +70,9 @@ public String getElementId() {
 
   @Override
   public boolean matches(Executable element) {
-    return element instanceof Constructor && new ConstructorIdentifier((Constructor) element).equals(this);
+    return element instanceof Constructor
+        && getElementName().equals(element.getDeclaringClass().getSimpleName())
+        && matchesArguments(element.getParameterTypes());
```

Comment:
this does not look good, the equality of the identifier should be enough, why falling back to each part separately?